language/move-lang/tests/move_check/typing/number_literal_too_large.exp

```diff
@@ -1,98 +1,64 @@
+  │
+6 │         (340282366920938463463374607431768211454: u8);
+  │          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  -- Expected a literal of type 'u8', but the value is too large.
+  │          │                                         
+  │          Invalid numerical literal
+  │          Annotating the literal might help inference: '340282366920938463463374607431768211454u128'
 
-   ┌── tests/move_check/typing/number_literal_too_large.move:5:10 ───
-   │
- 5 │         (340282366920938463463374607431768211454: u64);
-   │          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Invalid numerical literal
-   ·
- 5 │         (340282366920938463463374607431768211454: u64);
-   │                                                   --- Expected a literal of type 'u64', but the value is too large.
-   ·
- 5 │         (340282366920938463463374607431768211454: u64);
-   │          --------------------------------------- Annotating the literal might help inference: '340282366920938463463374607431768211454u128'
-   │
+error[E04021]: invalid number after type inference
+  ┌─ tests/move_check/typing/number_literal_too_large.move:8:10
+  │
+8 │         (18446744073709551616: u64);
+  │          ^^^^^^^^^^^^^^^^^^^^  --- Expected a literal of type 'u64', but the value is too large.
+  │          │                      
+  │          Invalid numerical literal
+  │          Annotating the literal might help inference: '18446744073709551616u128'
 
-error: 
+error[E04021]: invalid number after type inference
+  ┌─ tests/move_check/typing/number_literal_too_large.move:9:10
+  │
+9 │         (18446744073709551616: u8);
+  │          ^^^^^^^^^^^^^^^^^^^^  -- Expected a literal of type 'u8', but the value is too large.
+  │          │                      
+  │          Invalid numerical literal
+  │          Annotating the literal might help inference: '18446744073709551616u128'
 
-   ┌── tests/move_check/typing/number_literal_too_large.move:6:10 ───
+error[E04021]: invalid number after type inference
+   ┌─ tests/move_check/typing/number_literal_too_large.move:11:10
    │
- 6 │         (340282366920938463463374607431768211454: u8);
-   │          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Invalid numerical literal
-   ·
- 6 │         (340282366920938463463374607431768211454: u8);
-   │                                                   -- Expected a literal of type 'u8', but the value is too large.
-   ·
- 6 │         (340282366920938463463374607431768211454: u8);
-   │          --------------------------------------- Annotating the literal might help inference: '340282366920938463463374607431768211454u128'
-   │
-
-error: 
+11 │         (18446744073709551615: u8);
+   │          ^^^^^^^^^^^^^^^^^^^^  -- Expected a literal of type 'u8', but the value is too large.
+   │          │                      
+   │          Invalid numerical literal
+   │          Annotating the literal might help inference: '18446744073709551615u64'
```

Comment:
Commenting for a later date/PR: this help message seems incorrect (if I tried to do what it says I would still get the same error). I'm guessing this is stemming from not knowing whether the literal has been annotated or not in typing?

Reply:
Let me take a look

Reply:
I think if changing it will then lead to a different error that will pinpoint the problem, then not adding the additional complexity of the special case  and keeping it the way it is today seems like the better option to me.

Reply:
So, land as is? and I won't make the change above?